pkg/kubelet/dockertools/manager.go

```diff
@@ -1748,16 +1772,69 @@ func (dm *DockerManager) clearReasonCache(pod *api.Pod, container *api.Container
 	dm.reasonCache.Remove(pod.UID, container.Name)
 }
 
+// TODO (random-liu) Move this to container/runtime.go
+
+// SyncAction indicates different kind of actions in SyncPod(). Now there are only actions
+// about start/kill container and setup/teardown network.
+type SyncAction int
+
+const (
+	StartContainer SyncAction = iota
+	KillContainer
+	SetupNetwork
+	TeardownNetwork
+	// Add more later if needed
+)
+
+// SyncResult is the result of sync action.
+type SyncResult struct {
+	// The associated action of the result
+	Action SyncAction
+	// The target of the action, now the target can only be:
+	//  * Container: Target should be container name
+	//  * Network: Target is useless now, we just set it as pod full name now
+	Target interface{}
+	// Brief error reason
+	Error error
+	// Human readable error reason
+	Message string
+}
+
+func (r *SyncResult) Fail(err error, msg string) {
+	r.Error, r.Message = err, msg
+}
+
+// Now the PodSyncResult is just a slice holding all the SyncResult
+type PodSyncResult []*SyncResult
+
+func (p PodSyncResult) NewSyncResult(action SyncAction, target interface{}) *SyncResult {
```

Comment:
Because we may fail the result later, so we need keep the reference of the result.